bundles/net.tourbook/src/net/tourbook/ui/tourChart/action/ActionSelectInBetweenTimeSlices.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (C) 2020 Frédéric Bard
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110, USA
+ *******************************************************************************/
+package net.tourbook.ui.tourChart.action;
+
+import net.tourbook.chart.IChartContextProvider;
+import net.tourbook.chart.SelectionChartInfo;
+import net.tourbook.common.util.Util;
+import net.tourbook.tour.TourManager;
+import net.tourbook.ui.views.tourDataEditor.TourDataEditorView;
+
+import org.eclipse.jface.action.Action;
+
+public class ActionSelectInBetweenTimeSlices extends Action {
+
+   private final IChartContextProvider _chartContextProvider;
+
+   public ActionSelectInBetweenTimeSlices(final IChartContextProvider chartContextProvider, final String text) {
+
+      super(text);
+
+      _chartContextProvider = chartContextProvider;
+   }
+
+   @Override
+   public void run() {
+
+      // make sure the tour editor does not contain a modified tour
+      if (TourManager.isTourEditorModified()) {
+         return;
+      }
+
+      //Opens the TourDataEditorView
+      Util.showView(TourDataEditorView.ID, true);
+
+      final TourDataEditorView tourDataEditorView = TourManager.getTourDataEditor();
+      final SelectionChartInfo selectionChartInfo = _chartContextProvider.getChart().getChartInfo();
+
+      if (selectionChartInfo == null || tourDataEditorView == null) {
+         return;
+      }
+
+      //Opens the TimeSlice tab
+      tourDataEditorView.selectTimeSlicesTab();
+
+      tourDataEditorView.setRowEditModeEnabled(true);
+
+      tourDataEditorView.selectTimeSlice_InViewer(selectionChartInfo.leftSliderValuesIndex, selectionChartInfo.rightSliderValuesIndex);
+
+      //We recreate the viewer to take into account the new selection
+      tourDataEditorView.recreateViewer();
```

Comment:
Why not putting all tour editor methods directly into one method in the tour editor, e.g. `actionSelectInBetweenTimeSlices()`?

This also have the advantage to not set some methods public.

Why are you using `tourDataEditorView.recreateViewer();` this is a very expensive methods and have a signifant delay when you have >10'000 time slices or a >3h tour?

Reply:
Instead of an action to select time slices in the tour editor, this feature may also be a flag that the movement of the sliders is selecting immediately the time slices, either one (how it currently works) or all between.

Reply:
Instead of a checkbox action, a checkbox in the chart options would be a better solution for my previous comment

![chart-option](https://user-images.githubusercontent.com/1283445/85109030-fc47b480-b210-11ea-975d-6611db942f2d.png)

Reply:
>Instead of a checkbox action, a checkbox in the chart options would be a better solution for my previous comment

I had that originally but I didn't like it because it might use unnecessary extra CPU cycles if the user keeps it checked and moves the sliders but doesn't need to select the time slices.
Hence why I changed to an action that a user clicks when it's really needed.

Let me know what you prefer.

Reply:
> Why are you using tourDataEditorView.recreateViewer(); this is a very expensive methods and have a signifant delay when you have >10'000 time slices or a >3h tour?

Because otherwise the new selected time slices are not selected in the viewer. So this is the only solution I found to refresh the viewer and see the selected new time slices.

Reply:
> I had that originally but after thinking about it I didn't like it because it might use unnecessary extra CPU cycles if the user keeps it checked and moves the sliders but doesn't need to select the time slices.
> Hence why I changed to an action that a user clicks when it's really needed.
> 
> Let me know what you prefer.

In MT there are some cases where this situation occures but it could be solved relative easily by waiting for a few ms and do the action when the user has stopped, e.g. you can find a similar case here

https://github.com/wolfgang-ch/mytourbook/blob/150ab2b37685edb7468d2b55c85f4e0bc1a35bab/bundles/net.tourbook/src/net/tourbook/map2/view/Map2View.java#L2064-L2100

When searching MT for `timerExec()` there will be some findings, however not only for this situation. The most prominent (but more complicated) case is the drawing of the tour chart which is also delayed and do not happen when resizing the chart window.

Reply:
> > Why are you using tourDataEditorView.recreateViewer(); this is a very expensive methods and have a signifant delay when you have >10'000 time slices or a >3h tour?
> 
> Because otherwise the new selected time slices are not selected in the viewer. So this is the only solution I found to refresh the viewer and see the selected new time slices.
> Ideas ?

I think it should be possible and will try to solve it when you've fixed the above issue.

Reply:
> 
> 
> > > Why are you using tourDataEditorView.recreateViewer(); this is a very expensive methods and have a signifant delay when you have >10'000 time slices or a >3h tour?
> > 
> > 
> > Because otherwise the new selected time slices are not selected in the viewer. So this is the only solution I found to refresh the viewer and see the selected new time slices.
> > Ideas ?
> 
> I think it should be possible and will try to solve it when you've fixed the above issue.

I am confused : If I implement the "_above issue_" like you say (i.e. : _a checkbox in the chart options would be a better solution for my previous comment_) then I don't need to call this `tourDataEditorView.recreateViewer()` as when the user moves the sliders, the time slices will be selected in the background but the viewer will not need to be recreated.
Does that make sense ?

Reply:
>then I don't need to call this `tourDataEditorView.recreateViewer()` as when the user moves the sliders, the time slices will be selected in the background but the viewer will not need to be recreated.

When you implement this feature with the `timerExec()` solution then a recreation of the viewer is not necessary

Reply:
Addressed in this [PR](https://github.com/wolfgang-ch/mytourbook/pull/192)